docs/specifications/tests/DNSSEC-TP/dnssec17.md

```diff
@@ -0,0 +1,213 @@
+          in the answer section, then add the name server IP and the
+          CDS RRset to the *CDS RRsets* set. Also include any assciated
+          RRSIG records.
+    2. Send the CDNSKEY query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server
+          IP.
+       2. Else, if AA bit is not set in the DNS response, then go to
+          next name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then
+          go to next name server IP.
+       4. Else, if the DNS response contains at least one CDNSKEY
+          record in the answer section, then add the name server IP and
+          the CDNSKEY RRset from the answer section to the 
+          *CDNSKEY RRsets* set. Also include any assciated RRSIG records.
+    3. Send the DNSKEY query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server
+          IP.
+       2. Else, if AA bit is not set in the DNS response, then go to
+          next name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then
+          go to next name server IP.
+       4. Else, if the DNS response contains at least one DNSKEY
+          record in the answer section, then add the name server IP and
+          the CDNSKEY RRset from the answer section to the 
+          *DNSKEY RRsets* set.
+    4. Go to next name server IP.
+
+11. If neither of the *CDS RRsets* and *CDNSKEY RRsets* sets,
+    respectively, has any RRset then terminate this test case.
+
+12. If the *DNSKEY RRsets* is empty, then terminate this test case.
+
+13. For each name server IP in the *CDS RRsets* set do:
+
+    1. Extract the RRSIG records for the CDS RRset.
+    2. Extract the DNSKEY from the *DNSKEY RRsets* for the same name
+       server IP.
+    3. For each DS record in *DS Records* do:
+       1. If the DS record does not point at a DNSKEY record then add
+          name server IP and DS key tag to the 
+          *DS No Match CDS RRSIG*.
+       2. Else, if the DNSKEY that the DS record points at does not
+          match any RRSIG for CDS RRset then add name server IP and
+          DS key tag to the *DS No Match CDS RRSIG*.
+    4. Go to next name server IP address.
+
+14. For each name server IP in the *CDNSKEY RRsets* set do:
+
+    1. Extract the RRSIG records for the CDNSKEY RRset.
+    2. Extract the DNSKEY from the *DNSKEY RRsets* for the same name
+       server IP.
+    3. For each DS record in *DS Records* do:
+       1. If the DS record does not point at a DNSKEY record then add
+          name server IP and DS key tag to the 
+          *DS No Match CDNSKEY RRSIG*.
+       2. Else, if the DNSKEY that the DS record points at does not
+          match any RRSIG for CDNSKEY RRset then add name server IP and
+          DS key tag to the *DS No Match CDNSKEY RRSIG*.
+    4. Go to next name server IP address.
+
```

Comment:
Overall question: How is "match" defined for keys? Only matching alg/keytag or are signatures actually verified?

Reply:
Presently is probably only verifying key tag.

Reply:
I think this should be validate.

Reply:
I agree, and that is an improvement that maybe can be included for this release for this test case.